aws/templates/solution/solution_userpool.ftl

```diff
@@ -1,209 +1,301 @@
+        [#assign smsInviteMessage = ""] 
+        [#if (appSettingsObject.UserPool.SMSInviteMessage)?has_content ]
+            [#assign smsInviteMessage = appSettingsObject.UserPool.SMSInviteMessage ]
+        [/#if]
+
+        [#if ( (configuration.MFA) || ( configuration.verifyPhone)) ]
+            [#if deploymentSubsetRequired("iam", true) &&
+            isPartOfCurrentDeploymentUnit(userPoolId) ]
+
+                [@createRole
+                    mode=listMode
+                    id=userPoolRoleId
+                    trustedServices=["cognito-idp.amazonaws.com" ]
+                    policies=
+                        [
+                            getPolicyDocument(
+                                snsPublishPermission(),
+                                "smsVerification" 
+                            )
+                        ]
+                /]
+
+                [#assign phoneSchema = getUserPoolSchemaObject( 
+                                            "phone_number",
+                                            "String",
+                                            true,
+                                            true)]
+                [#assign schema = schema + [ phoneSchema ]]
+
+                )]
+
+            [/#if]
+
+            [#assign smsConfig = getUserPoolSMSConfiguration( getReference(userPoolRoleId, ARN_ATTRIBUTE_TYPE), userPoolName )]
+            [#assign smsVerification = true]
+        [/#if]
+
+        [#if configuration.verifyEmail || ( configuration.loginAliases.seq_contains("email") ) ]
+                [#assign emailSchema = getUserPoolSchemaObject( 
+                                            "email",
+                                            "String",
+                                            true,
+                                            true)]
+                [#assign schema = schema +  [ emailSchema ]]
+        [/#if]
+
+        [#list configuration.Links?values as link]
+            [#if link?is_hash]
+                [#assign linkTarget = getLinkTarget(occurrence, link, true) ]
+                [@cfDebug listMode linkTarget false /]
+
+                [#assign linkTargetCore = linkTarget.Core ]
+                [#assign linkTargetConfiguration = linkTarget.Configuration ]
+                [#assign linkTargetResources = linkTarget.State.Resources ]
+                [#assign linkTargetAttributes = linkTarget.State.Attributes ]
+
+                [#switch linkTargetCore.Type!""]
+                    [#case LAMBDA_FUNCTION_COMPONENT_TYPE ]
+
+                        [#-- Cognito Userpool Event Triggers --]
+                        [#switch link.Name ]
+                            [#case "createauthchallenge" ]
+                                [#assign userPoolTriggerConfig =  userPoolTriggerConfig +
+                                    attributeIfContent (
+                                        "CreateAuthChallenge",
+                                        linkTargetAttributes.ARN
```

Comment:
think we should be checking the Deployed property and remove the allowNotFound flag...